livechat/utils/ws_client.py

```diff
@@ -2,153 +2,96 @@
-        self.url = url
-        self.timeout = timeout
-        self.websocket: WebSocket = None
-        self.keep_alive = True
+    messages: List[dict] = []
+
+    def __init__(self, *args, **kwargs):
         logging.basicConfig()
         self.logger = logging.getLogger()
         self.logger.setLevel(logging.INFO)
+        super().__init__(*args, **kwargs)
+        self.on_message = on_message
 
-    def open(self, keep_alive: bool = True, origin: dict = None) -> None:
-        ''' Open WebSocket connection.
-                Args:
-                    keep_alive(bool): Bool which states if connection should be kept, by default sets to `True`.
-                    origin (dict): Specifies origin while creating websocket connection.
-        '''
+    def _wait_till_sock_connected(self, timeout: float = 3):
+        ''' Polls until `self.sock` is connected.
+            Args:
+                timeout (float): timeout value in seconds, default 3. '''
+        if timeout < 0:
+            raise TimeoutError('Timed out waiting for WebSocket to open.')
         try:
-            if origin:
-                self.websocket = websocket.create_connection(
-                    self.url,
-                    self.timeout,
-                    origin=origin,
-                    sslopt={'cert_reqs': ssl.CERT_NONE},
-                    enable_multithread=True)
-            else:
-                self.websocket = websocket.create_connection(
-                    self.url,
-                    self.timeout,
-                    sslopt={'cert_reqs': ssl.CERT_NONE},
-                    enable_multithread=True)
-        except (WebSocketException, WebSocketTimeoutException,
-                WebSocketProtocolException, WebSocketPayloadException,
-                WebSocketConnectionClosedException, WebSocketProxyException,
-                WebSocketBadStatusException,
-                WebSocketAddressException) as exception:
-            self.logger.critical(f'WebSocket Exception: {exception}')
-        else:
-            if keep_alive:
-                self.keep_alive = True
-                threading.Thread(target=self._keep_ws_alive).start()
+            assert self.sock.connected
+            return
+        except (AttributeError, AssertionError):
+            sleep(0.1)
+            return self._wait_till_sock_connected(timeout=timeout - 0.1)
 
-    def close(self) -> None:
-        ''' Close WebSocket connection. '''
-        self.keep_alive = False
-        if self.websocket.connected:
-            try:
-                self.websocket.close()
-            except WebSocketConnectionClosedException as error:
-                self.logger.critical(
-                    f'Exception caught while closing the WebSocket: {error}.')
+    def open(self,
+             origin: dict = None,
+             timeout: float = 3,
+             keep_alive: bool = True):
+        ''' Open websocket connection and keep running forever.
```

Comment:
`Opens` because you use `Polls` in `_wait_till_sock_connected` :) Applicable to similar occurrences